pkg/storaged/resize.jsx

```diff
@@ -0,0 +1,441 @@
+    }
+
+    const usage = get_active_usage(client, block && !to_fit && info.shrink_needs_unmount ? block.path : null,
+                                   _("shrink"));
+
+    if (usage.Blocking) {
+        dialog_open({
+            Title: cockpit.format(_("$0 is in use"), name),
+            Body: BlockingMessage(usage)
+        });
+        return;
+    }
+
+    let shrink_size;
+    let size_fields = [];
+    if (!to_fit) {
+        size_fields = [
+            SizeSlider("size", _("Size"),
+                       {
+                           value: orig_size,
+                           max: orig_size,
+                           round: round_size,
+                       })
+        ];
+    } else {
+        const crypto = client.blocks_crypto[block.path];
+        const cleartext = client.blocks_cleartext[block.path];
+        let content_path = null;
+        let crypto_overhead = 0;
+
+        if (crypto) {
+            if (crypto.MetadataSize !== undefined && cleartext) {
+                content_path = cleartext.path;
+                crypto_overhead = crypto.MetadataSize;
+            }
+        } else {
+            content_path = block.path;
+        }
+
+        const fsys = client.blocks_fsys[content_path];
+        if (fsys)
+            shrink_size = fsys.Size + crypto_overhead;
+
+        const vdo = client.legacy_vdo_overlay.find_by_backing_block(client.blocks[content_path]);
+        if (vdo)
+            shrink_size = vdo.physical_size + crypto_overhead;
+
+        const stratis_bdev = client.blocks_stratis_blockdev[content_path];
+        if (stratis_bdev)
+            shrink_size = Number(stratis_bdev.TotalPhysicalSize) + crypto_overhead;
+
+        if (shrink_size === undefined) {
+            console.warn("Couldn't determine size to shrink to.");
+            return;
+        }
+    }
+
+    let recovered_passphrase;
+    let passphrase_fields = [];
+    if (block && block.IdType == "crypto_LUKS" && block.IdVersion == 2)
+        passphrase_fields = existing_passphrase_fields(_("Resizing an encrypted filesystem requires unlocking the disk. Please provide a current disk passphrase."));
+
+    if (usage.length == 0 && size_fields.length + passphrase_fields.length === 0) {
+        return lvol_or_part_and_fsys_resize(client, lvol_or_part, shrink_size, false, null);
+    }
+
+    const dlg = dialog_open({
+        Title: title,
+        Teardown: TeardownMessage(usage),
+        Fields: size_fields.concat(passphrase_fields),
+        Action: {
+            Title: _("Shrink"),
+            action: function (vals) {
+                return teardown_active_usage(client, usage)
+                        .then(function () {
+                            return (lvol_or_part_and_fsys_resize(client, lvol_or_part,
+                                                                 to_fit ? shrink_size : vals.size,
+                                                                 to_fit ? false : info.shrink_needs_unmount,
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19385-20230926-092314-7463d2dd-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)